kv: acceptance/gossip/peerings flaked

See flake in CI [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=8469572&problemId=11989).

This issue depends on resolving #74714

Jira issue: CRDB-23944

```
* ERROR: ERROR: server startup failed: cockroach server exited with error: result is ambiguous: error=grpc: error reading from server: read tcp 127.0.0.1:56366->127.0.0.1:26263: read: connection reset by peer [code 14/Unavailable] [exhausted]
* SQLSTATE: 40003
*
ERROR: server startup failed: cockroach server exited with error: result is ambiguous: error=grpc: error reading from server: read tcp 127.0.0.1:56366->127.0.0.1:26263: read: connection reset by peer [code 14/Unavailable] [exhausted]
SQLSTATE: 40003
Failed running "start"
```

and I came here to report another one:

https://teamcity.cockroachdb.com/viewLog.html?buildId=8648958&buildTypeId=Cockroach_BazelEssentialCi

```
I230209 14:47:30.888468 342 kv/kvserver/store_remove_replica.go:150 â‹® [T1,n1,s1,r58/1:â€¹/{Table/57-Max}â€º,raft] 108  removing replica r58/1
I230209 14:47:31.041859 1 1@cli/start.go:900 â‹® [T1,n1] 109  initiating graceful shutdown of server
I230209 14:47:31.042094 1 1@cli/start.go:961 â‹® [T1,n1] 110  too early to drain; used hard shutdown instead
E230209 14:47:31.043987 1 1@cli/clierror/check.go:35 â‹® [T1] 111  server startup failed: cockroach server exited with error: result is ambiguous: error=ba: â€¹Put [/Table/46/1/1/0,/Min), EndTxn(parallel commit) [/Table/46/1/1/0], [txn: eca4807c], [can-forward-ts]â€º RPC error: grpc: â€¹node waiting for init; /cockroach.roachpb.Internal/Batch not availableâ€º [code 14/Unavailable] [exhausted]
```

In other words, while we're sleeping here, 

https://github.com/cockroachdb/cockroach/blob/f2bbad960cd3c9e0a46633374050fb32bb648137/pkg/cmd/roachtest/tests/gossip.go#L303-L306

the node crashes during startup. (There isn't a `monitor` running in this test that would expose this more directly).

--- a/pkg/cmd/roachtest/tests/acceptance.go
+++ b/pkg/cmd/roachtest/tests/acceptance.go
@@ -31,7 +31,7 @@ func registerAcceptance(r registry.Registry) {
 		registry.OwnerKV: {
 			{name: "decommission-self", fn: runDecommissionSelf},
 			{name: "event-log", fn: runEventLog},
-			{name: "gossip/peerings", fn: runGossipPeerings, skip: "flaky test. tracked in #96091"},
+			{name: "gossip/peerings", fn: runGossipPeerings},
 			{name: "gossip/restart", fn: runGossipRestart},
 			{
 				name:              "gossip/restart-node-one",

